### PR TITLE
[WIP] 98/build form front

### PR DESCRIPTION
**Pull request by dgrcode**
- [x] Submit
 - [x] Review

### CODE REVIEW

File: packages/react-app/src/views/BuildsListView.jsx [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+} from "@chakra-ui/react";
 import BuildCard from "../components/BuildCard";
-import { getAllBuilds } from "../data/api";
+import { getAllBuilds, getBuildSubmitSignMessage, postBuildSubmit } from "../data/api";
 import useCustomColorModes from "../hooks/useCustomColorModes";
 
-export default function BuildsListView() {
+export default function BuildsListView({ userProvider }) {
   const [builds, setBuilds] = useState([]);
+  const [isSubmitModalOpen, setIsSubmitModalOpen] = useState(false);
```
Comment (carletex) [inline packages/react-app/src/views/BuildsListView.jsx:29]: Just FYI, Chakra has a fancy useDisclosure hook to handle open / close / toggle scenarios.

https://chakra-ui.com/docs/hooks/use-disclosure

File: packages/react-app/src/views/BuildsListView.jsx [hunk 2]
```diff
@@ -16,17 +46,156 @@ export default function BuildsListView() {
...
+              <Input
+                id="buildUrl"
+                placeholder="https://..."
+                value={buildUrl}
+                onChange={evt => setBuildUrl(evt.target.value)}
+              />
+              <FormErrorMessage>This field is required</FormErrorMessage>
+            </FormControl>
+            <FormControl mb={4} isInvalid={errors.imageUrl}>
+              <FormLabel htmlFor="imageUrl">Build name</FormLabel>
```
Comment (carletex) [inline packages/react-app/src/views/BuildsListView.jsx:185]: This should be Image URL

Comment (carletex) [reply packages/react-app/src/views/BuildsListView.jsx]: Fixed on https://github.com/moonshotcollective/scaffold-directory/pull/109/commits/b0c044228782cc3fb9dcec574d52d55018fec19e

**Comment by carletex** [comment]
@dgrcode Everything looks great!

Let me know when the review part is ready so I can merge it.

File: packages/react-app/src/views/BuildsListView.jsx (hunk 1 above)
Comment (dgrcode) [reply packages/react-app/src/views/BuildsListView.jsx:29]: Awesome. I just found two usages in our app, so I'll update the code for consistency.

**Comment by dgrcode** [comment]
To make things easier to read, I propose to update the following property names of the builds
 - address -> userAddress
 - branch -> branchUrl
 - desc -> description
 - image -> imageUrl

 (I'd keep `id`, `isDraft` and `name` the same)

Issues:
- [ ] When a build is rejected, a `null` build is created. This happens locally, haven't checked in firebase
- [ ] Handle builds without images

**Pull request merged**